araproc/analysis/correlation_snr.py

```diff
@@ -0,0 +1,254 @@
+    
+    Parameters
+    ----------
+    None
+    
+    Returns
+    -------
+    pairs: list
+       List of rf channel pairs.
+    paits_V: list
+       List of VPol rf channel pairs.
+    paits_H: list
+       List of HPol rf channel pairs.
+    """
+
+    rf_chs = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+
+    pairs = list(itertools.combinations(rf_chs, 2))
+    pairs_V = list(itertools.combinations(rf_chs[:8], 2))
+    pairs_H = list(itertools.combinations(rf_chs[8:], 2))
+
+    return pairs, pairs_V, pairs_H
+
+
+
+def pairing(channels):
+    
+    """
+    Makes channel pairs.
+    
+    Parameters
+    ----------
+    channels: list
+       List of given channels.
+
+    Returns
+    -------
+    pairs: list
+       List of all possible pair combinations of given channels.
+    """
+ 
+    pairs = list(itertools.combinations(channels, 2))
+    
+    return pairs
+
+
+
+def get_corr(y1, y2):
+    
+    """
+    Calculates normalized correlation between two waveforms, y1 and y2.
+    
+    Parameters
+    ----------
+    y1: list
+       List of values of waveform 1.
+    y2: list
+       List of values of waveform 2.
+    
+    Returns
+    -------
+    corr: array
+       Array of values of cross correlation function.  
+    """
+
+    _, y1 = wfu.tgraph_to_arrays(y1)
+    _, y2 = wfu.tgraph_to_arrays(y2)
+    
+    n = len(y1)
+    corr = co(y2, y1, mode = 'same')/np.sqrt(co(y1, y1, mode = 'same')[int(n/2)]*co(y2, y2, mode = 'same')[int(n/2)])    
+
+    return corr
+
+
+
+def get_corr_snr(first_ch, second_ch):
+
+    """
+    Calculates channel-pair correlation SNR
+    
+    Parameters
+    ----------
+    first_ch: array 
+       Array of a given channel waveform.
+    second_ch: array 
+       Array of another given channel waveform.
+     
+    Returns
+    -------
+    corr_snr: float
+       Channel pair correlation SNR.
+    """
+    
+    corr = get_corr(first_ch, second_ch)
+    corr_snr = snr.get_snr(corr)
+
+    return corr_snr
+
+
+def get_avg_corr_snr(wave_bundle, excluded_channels = []):
+
+    """
+    Makes useful dictionaries for channel-pair correlation SNR and their average.
+
+    Parameters
+    ----------
+    wave_bundle: dict of TGraphs
+        Dictionary of waveform TGraphs to be averaged.
+    excluded_channels: list
+        List of dictionary keys to exclude from average.
+
+    Returns
+    -------
+    avg_corr_snr : float
+        Average channel-pair correlation SNR.
+    avg_vpol_corr_snr : float
+        The VPol average channel-pair correlation SNR.
+    avg_hpol_corr_snr : float
+        The HPol average channel-pair correlation SNR.
+    """
+
+
+    chans = list(wave_bundle.keys())
+
+    AllPol_dict, VPol_dict, HPol_dict = {}, {}, {}
+    corr_snr, VPol_corr_snr, HPol_corr_snr = {}, {}, {}
+
+    for chan in chans:
+
+      if(chan in excluded_channels):
+        continue
+
+      AllPol_dict.update({chan: wave_bundle[chan]})
+
+      if chan in chans[:8]:
```

Comment:
we should check the channel numbers, what if chans doesn't include the channels you expect? or in the right order?

Reply:
Right! So removing 

`if(chan in excluded_channels):
        continue`
        
would solve the issue. Later excluded channels can be discarded. As far channel mapping, the RF channels are same for all stations, no? I thought, when you call getGraphFromRFChan, it loads the correctly mapped Electric Channels.

Reply:
Sorry i mean more what is the first 8 entries in `chans` aren't vpols? like what if `chans=[1, 14, 3, 13, ...]` for example